tests/foreman/ui_airgun/test_rhai.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+from fauxfactory import gen_string
+from nailgun import entities
+
+from robottelo import manifests
+from robottelo.api.utils import upload_manifest as up_man
+from robottelo.constants import DEFAULT_SUBSCRIPTION_NAME, DISTRO_RHEL7
+from robottelo.vm import VirtualMachine
+
+
+pytestmark = pytest.mark.usefixtures("upload_manifest", "attach_subscription")
+
+
+@pytest.fixture(scope="module")
+def org():
+    org = entities.Organization(name="insights_{0}".format(
+        gen_string("alpha", 6))).create()
+    yield org
+    org.delete()
+
+
+@pytest.fixture
+def set_context(autosession, org):
+    autosession.organization.select(org.name)
+
+
+@pytest.fixture(scope="module")
+def upload_manifest(org):
+    with manifests.clone() as manifest:
+        up_man(org.id, manifest.content)
+
+
+@pytest.fixture(scope="module")
+def activation_key(org):
+    ak = entities.ActivationKey(
+        auto_attach=True,
+        content_view=org.default_content_view.id,
+        environment=org.library.id,
+        name=gen_string("alpha"),
+        organization=org
+    ).create()
+    yield ak
+    ak.delete()
+
+
+@pytest.fixture(scope="module")
+def attach_subscription(org, activation_key):
+    for subs in entities.Subscription(organization=org).search():
+        if subs.read_json()["product_name"] == DEFAULT_SUBSCRIPTION_NAME:
+            # "quantity" must be 1, not subscription["quantity"]. Greater
+            # values produce this error: "RuntimeError: Error: Only pools
+            # with multi-entitlement product subscriptions can be added to
+            # the activation key with a quantity greater than one."
+            activation_key.add_subscriptions(data={
+                "quantity": 1,
+                "subscription_id": subs.id,
+            })
+            break
+
+
+@pytest.fixture(scope="module")
+def vm(activation_key, org):
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+        vm.configure_rhai_client(
+            activation_key=activation_key.name,
+            org=org.label,
+            rhel_distro=DISTRO_RHEL7)
+        yield vm
+
+
+def test_positive_register_client_to_rhai(vm, autosession, set_context):
+    """Check client registration to redhat-access-insights service.
+
+    :id: 9815151e-d50d-4160-9d29-ae7c89422e18
+
+    :expectedresults: Registered client should appear in the Systems sub-
+        menu of Red Hat Access Insights
+    """
+
+    try:
+        assert autosession.insights_inventory.search(vm.hostname)
+        result = autosession.insights_inventory.total_systems
+        assert result == "1", "Registered clients are not listed"
+    except AssertionError:
```

Comment:
Another idea for you to reduce indentation and copy/pasted code for every single test - use `addfinalizer()` in `vm` fixture. If you want to keep logs of failed tests only - you could combine that with https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
Such way seems more logical to me - test won't take the responsibility of vm teardown, vm teardown will handle logs by its own.